src/sentry/tasks/post_process.py

```diff
@@ -359,6 +361,28 @@ def post_process_group(
             event_processing_store.delete_by_key(cache_key)
 
 
+def process_categories(event):
+    from sentry.models.groupcategory import GroupCategory
+
+    for category in _extract_categories_from_event(event):
+        GroupCategory.objects.create(group_id=event.group_id, category=category)
```

Comment:
Creating up to 4 `GroupCategory` objects feels wrong, I wonder if there's a way to accomplish the same functionality with something like a BitField so we can determine the active categories with a single Entry in the Category table?

Reply:
I am worried here that we'll end up processing the same event multiple times, and end up with duplicate GroupCategory entries. Should we maybe have a unique_together index on group + category to explicitly disallow this?

So, this create would be relatively quick (at least compared to say .filter in most cases), but I still think it'd be wise to offload some of this work. 

I'm thinking that employing debouncing or even implementing this as a task might be overkill and not much to gain. I do think there would be value in just instantiating GroupCategory objects, storing them in a list, and using `GroupCategory.objects.bulk_create` instead though.

Reply:
I was thinking about this more, if we just want categories and the docstring on the `_extract_categories_from_event` says that `it is fast` depending on the usecase we might be able to extract categories at runtime and skip the table entirely